src/gui/qgsattributedialog.cpp

```diff
@@ -23,7 +23,7 @@
 #include "qgssettings.h"
 
 QgsAttributeDialog::QgsAttributeDialog( QgsVectorLayer *vl, QgsFeature *thepFeature, bool featureOwner, QWidget *parent, bool showDialogButtons, const QgsAttributeEditorContext &context )
-  : QDialog( parent )
+  : QDialog( parent, Qt::Window )
```

Comment:
Out of curiosity, what's that for?

Reply:
to make it non modal (like in [QgsAttributeTableDialog](https://github.com/troopa81/QGIS/blob/fix_relationeditor_crash_context/src/app/qgsattributetabledialog.h#L56)) so the user can still click on the canvas.

Reply:
What about using `Qt::Tool`?

Using this approach will likely make it disappear behind the main window (leaving people with 20 open attribute dialogs - just like they now tend to open the attribute table 20 times)

Reply:
> Using this approach will likely make it disappear behind the main window 

That's the exact reason of this modification. It's because of this [PR](https://github.com/qgis/QGIS/pull/32528). I want to be able to raise the canvas so user can digitize a feature and then raise back the attribute dialog. 

> (leaving people with 20 open attribute dialogs - just like they now tend to open the attribute table 20 times)

I understand your concern, but I can't see a better way to solve this UI issue. Maybe, we could let the window on top of the canvas but the user can still digitize in the canvas (he has to move the window on the side though) but it's not really user friendly.

Reply:
A better approach would be

- hide the window
- set the map tool
- show instructions in the message bar
- when done re-raise the window
- when the map tool is deactivated re-raise the window

Reply:
Indeed, it seems the best approach, thanks for the hint, I made the modifications